omisego-sdk/src/test/kotlin/co/omisego/omisego/websocket/listener/FilteredCustomEventListenerTest.kt

```diff
@@ -0,0 +1,94 @@
+import org.junit.Test
+
+class FilteredCustomEventListenerTest {
+    private val mockUserSocketEvent: SocketEvent<TransactionConsumption> = mock {
+        on { socketReceive } doReturn mock<SocketReceive<TransactionConsumption>>()
+        on { socketReceive.topic } doReturn "user"
+    }
+    private val mockTransactionRequestSocketEvent: SocketEvent<TransactionConsumption> = mock {
+        on { socketReceive } doReturn mock<SocketReceive<TransactionConsumption>>()
+        on { socketReceive.topic } doReturn "transaction_request"
+    }
+    private val mockTransactionConsumptionSocketEvent: SocketEvent<TransactionConsumption> = mock {
+        on { socketReceive } doReturn mock<SocketReceive<TransactionConsumption>>()
+        on { socketReceive.topic } doReturn "transaction_consumption"
+    }
+    private val mockUser: User = mock {
+        on { socketTopic } doReturn SocketTopic("user")
+    }
+    private val mockTransactionRequest: TransactionRequest = mock {
+        on { socketTopic } doReturn SocketTopic("transaction_request")
+    }
+    private val mockTransactionConsumption: TransactionConsumption = mock {
+        on { socketTopic } doReturn SocketTopic("transaction_consumption")
+    }
+
+    @Test
+    fun `TransactionRequestTopicListener should accept an event only if the topic is the same`() {
+        val listener = object : TransactionRequestTopicListener(mockTransactionRequest) {
+            override fun onTransactionConsumptionRequest(transactionConsumption: TransactionConsumption) {
+            }
+
+            override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+            }
+
+            override fun onTransactionConsumptionFinalizedFail(transactionConsumption: TransactionConsumption, apiError: APIError) {
+            }
+        }
+
+        listener.strategy shouldBeInstanceOf FilterStrategy.Topic::class.java
+        (listener.strategy as FilterStrategy.Topic).accept(mockTransactionRequestSocketEvent) shouldBe true
+        (listener.strategy as FilterStrategy.Topic).accept(mockTransactionConsumptionSocketEvent) shouldBe false
+        (listener.strategy as FilterStrategy.Topic).accept(mockUserSocketEvent) shouldBe false
+    }
+
+    @Test
+    fun `TransactionConsumptionTopicListener should accept an event only if the topic is the same`() {
+        val listener = object : TransactionConsumptionTopicListener(mockTransactionConsumption) {
+            override fun onTransactionConsumptionFinalizedSuccess(transactionConsumption: TransactionConsumption) {
+            }
+
+            override fun onTransactionConsumptionFinalizedFail(transactionConsumption: TransactionConsumption, apiError: APIError) {
+            }
+        }
+
+        listener.strategy shouldBeInstanceOf FilterStrategy.Topic::class.java
+        (listener.strategy as FilterStrategy.Topic).accept(mockTransactionRequestSocketEvent) shouldBe false
```

Comment:
Do you really need to cast the strategy for the `accept` verifications? I don't think so 🤔

Reply:
Yeah, it isn't need